src/main/java/org/openrewrite/kotlin/internal/KotlinParserVisitor.java

```diff
@@ -84,6 +84,7 @@ public class KotlinParserVisitor extends FirDefaultVisitor<J, ExecutionContext>
     private int cursor;
 
     private Map<Integer, ASTNode> nodes;
```

Comment:
We don't need the ASTNodes at all. Each cursor may map to an object that contains the end offset and the debugging name. So, we could save memory by not preserving the tree